Marlin/src/inc/SanityCheck.h

```diff
@@ -930,6 +930,17 @@ static_assert(X_MAX_LENGTH >= X_BED_SIZE && Y_MAX_LENGTH >= Y_BED_SIZE,
   #error "MESH_EDIT_GFX_OVERLAY requires AUTO_BED_LEVELING_UBL and a Graphical LCD."
 #endif
 
+#if ENABLED(G29_RETRY_AND_RECOVER)
+  #if !HAS_LEVELING
+    #error "G29_RETRY_AND_RECOVER requires automatic bed leveling"
```

Comment:
This option will simply be ignored when leveling is disabled. It's more convenient to ignore it, so users may disable leveling for testing purposes without also having to disable `G29_RETRY_AND_RECOVER` before they can recompile.